datavalidation.go

```diff
@@ -256,30 +281,33 @@ func (f *File) AddDataValidation(sheet string, dv *DataValidation) error {
 	if err != nil {
 		return err
 	}
+	f.mu.Lock()
+	defer f.mu.Unlock()
 	if nil == ws.DataValidations {
 		ws.DataValidations = new(xlsxDataValidations)
 	}
-	dataValidation := &xlsxDataValidation{
-		AllowBlank:       dv.AllowBlank,
-		Error:            dv.Error,
-		ErrorStyle:       dv.ErrorStyle,
-		ErrorTitle:       dv.ErrorTitle,
-		Operator:         dv.Operator,
-		Prompt:           dv.Prompt,
-		PromptTitle:      dv.PromptTitle,
-		ShowDropDown:     dv.ShowDropDown,
-		ShowErrorMessage: dv.ShowErrorMessage,
-		ShowInputMessage: dv.ShowInputMessage,
-		Sqref:            dv.Sqref,
-		Type:             dv.Type,
+	dataValidation := newXlsxDataValidation(dv)
+	ws.DataValidations.DataValidation = append(ws.DataValidations.DataValidation, dataValidation)
+	ws.DataValidations.Count = len(ws.DataValidations.DataValidation)
+	return err
+}
+
+// BatchAddDataValidation is a function supports batch append data validation to the worksheet, reference to AddDataValidation().
```

Comment:
Any performance benchmark for this? To keep the library core and easy, and follow the "typically one right way to do most things" principles, anyone can easily make a custom function to add multiple data validations by themselves: batch append data validations, batch append tables, etc., so I don't suggest adding these functions.

Reply:
**BatchAddDataValidation() is not added in the latest commit.**  Here are some results of the benchmark.
BatchAddDataValidation() reduces the average op time by 37.9% compared to AddDataValidation(), but has an additional memory footprint of 4.3% and 1 alloc. 

example:
go test -benchmem -run=^$ -bench DataValidation$  -benchtime=100000x -cpu=8
goos: linux
goarch: amd64
pkg: github.com/xuri/excelize/v2
cpu: 11th Gen Intel(R) Core(TM) i5-11320H @ 3.20GHz
BenchmarkAddDataValidation-8              100000             33783 ns/op           20920 B/op        300 allocs/op
BenchmarkBatchAddDataValidation-8         100000             19289 ns/op           21815 B/op        301 allocs/op
PASS
ok      github.com/xuri/excelize/v2     5.465s